modules/tinymce/src/core/main/ts/fmt/RemoveFormat.ts

```diff
@@ -174,22 +175,39 @@ const removeNode = (ed: Editor, node: Node, format: Format) => {
   dom.remove(node, true);
 };
 
-const removeFormatInternal = (ed: Editor, format: Format, vars?: FormatVars, node?: Node, compareNode?: Node): RemoveFormatAdt => {
-  let stylesModified: boolean;
+// Attributes or styles can be either an array of names or an object containing name/value pairs
+const processFormatAttrOrStyle = (name: string | number, value: FormatAttrOrStyleValue | string, vars: FormatVars | undefined) => {
+  // Indexed array
+  if (Type.isNumber(name)) {
+    return {
+      name: value as string,
+      value: null
+    };
+  } else {
+    return {
+      name,
+      value: FormatUtils.replaceVars(value, vars)
+    };
+  }
+};
```

Comment:
Abstracted this out to make it a little clearer what's going on and to remove some duplication between the styles/attributes processing.